### PR TITLE
GPII-2091: Provisioning scripts fixed

### PR DESCRIPTION
**Pull request by kaspermarkus**
Now properly provisions and builds everything (including windows specific stuff).

"npm tests" will run all the tests (including windows + universal)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,12 +16,17 @@
     "grunt-shell": "1.3.0",
     "shelljs": "0.3.0"
   },
+  "scripts": {
+    "start": "electron main.js",
+    "test": "electron tests.js",
+    "postinstall": "cd ./node_modules/leveldown && node-gyp rebuild --target=1.4.1 --dist-url=https://atom.io/download/atom-shell"
```
Comment (michelled) [inline package.json:22]: I'm getting an error from npm install related to leveldown - is there something I need to do to make this work properly?

File: package.json [hunk 2]
```diff
@@ -20,12 +16,17 @@
...
+  },
   "dependencies": {
     "electron": "1.4.1",
     "request": "2.69.0",
-    "infusion": "2.0.0-dev.20161027T155641Z.0bcd507",
-    "windows": "git://github.com/GPII/windows.git#24cdf9164d0696a1617db250573e205b4568385b",
-    "universal": "git://github.com/GPII/universal.git#b58d1859e4d53303a1fdc4d74640e0c43af345e5",
+    "infusion": "git://github.com/amb26/infusion.git#240c00be023732ac8dcc193a57bc1b46c58d2de7",
```
Comment (michelled) [inline package.json:27]: Why do we need to depend on Antranig's repo now?

File: package.json [hunk 3]
```diff
@@ -20,12 +16,17 @@
...
+    "windows": "git://github.com/kaspermarkus/windows.git#04963330620af83baa03f92a227da7f4616e7aa1",
```
Comment (michelled) [inline package.json:28]: Is there a pull request related to this commit? Perhaps we can usher it through review.

File: README.md [hunk 4]
```diff
@@ -2,21 +2,37 @@
...
+## Using a Windows VM
 
-'npm start' (in the VM terminal)
+It is possible to provision a Windows VM for development and testing purposes. Please ensure you have met [these VM requirements](https://github.com/GPII/qi-development-environments/#requirements) before proceeding. 
 
+### Creating and removing the VM
 
-Running the Tests in a VM
--------------------------
+Once the requirements are meet, you can use the `vagrant up` command to create an instance of a [Windows 10 Evaluation VM](https://github.com/idi-ops/packer-windows) which will boot an instance of the Windows 10 VM, pull in the GPII Framework's npm dependencies, and then build it. Once it has finished building, either restart the VM, or open a Command Line or PowerShell and type the following command inside the VM: `refreshenv`
```
Comment (michelled) [inline README.md:14]: Typo here - 'Once the requirements are met'

Comment (michelled) [reply README.md:14]: I  think 'refreshenv' is no longer required.

File: provisioning/Build.ps1 [hunk 5]
```diff
@@ -0,0 +1,87 @@
+<#
```
Comment (michelled) [inline provisioning/Build.ps1:1]: How close is this to the scripts in the Window's repo? Should we come up with a strategy for reusing the bits that are the same?

File: package.json (hunk 1 above)
Comment (kaspermarkus) [reply package.json:22]: Nope - this postinstall script should be the thing needed.. Do you have a print of the error you're getting?

File: package.json (hunk 2 above)
Comment (kaspermarkus) [reply package.json:27]: This is a copy of the dependency from universal (the one referenced from the windows repo). @amb26 do you know what the state is of dependencies on your infusion branch?

File: package.json (hunk 3 above)
Comment (kaspermarkus) [reply package.json:28]: Yes - sorry for not posting this up front: https://github.com/GPII/windows/pull/107

File: provisioning/Build.ps1 (hunk 5 above)
Comment (kaspermarkus) [reply provisioning/Build.ps1:1]: I would say that it's heavily inspired by the windows scripts... The Provisioning.psm1 script is identical though. So you probably have a point

File: package.json (hunk 1 above)
Comment (michelled) [reply package.json:22]: The secret was 'vagrant provision' thanks @kaspermarkus! I suppose the different version of Node was causing the error I was seeing.

File: package.json (hunk 3 above)
Comment (michelled) [reply package.json:28]: Awesome! I commented on the pull request - hopefully it will get some attention.

File: provisioning/Build.ps1 (hunk 5 above)
Comment (michelled) [reply provisioning/Build.ps1:1]: it sounds like @gloob is taking this on - is that correct Ale?

**Comment by michelled** [comment]
@kaspermarkus are all the tests passing for you? I'm getting a failure in the processHandling tests.

**Pull request closed without merging**